kubernetes/template-generator/template_generator.py

```diff
@@ -0,0 +1,405 @@
+        :rtype: string or dict
+        """
+        service_template = {
+            "apiVersion": "v1",
+            "kind": "Service",
+            "metadata": {
+                "name": "leapp-{}-service".format(self.container_name),
+                "labels": {
+                    "name": "leapp-{}-label".format(self.container_name),
+                    }
+                },
+            "spec": {
+                "selector": {
+                    "name": "leapp-{}-label".format(self.container_name),
+                    },
+            }
+        }
+
+        if not self.external_ips:
+                service_template["spec"]["externalIPs"] = self.external_ips
+
+        if self.exposed_ports:
+            ports_template = {
+                "ports": []
+            }
+
+            for pmap in self.exposed_ports:
+                port = {
+                    "port": pmap[0],
+                    "targetPort": pmap[1],
+                    "protocol": "TCP",
+                    "name": "leapp-{}-{}-{}-port".format(self.container_name, pmap[0], pmap[1])
+                }
+                ports_template["ports"].append(port)
+
+            service_template["spec"].update(ports_template)
+
+        if yaml:
+            return self._dump_yaml(service_template)
+
+        return service_template
+
+    def generate_pod_template(self, yaml=True):
+        """
+        Generate a YAML template for kubernetes pod
+
+        :param yaml: if true, the output will be in yaml, else dict will be returned
+        :type yaml: bool
+
+        :return: a template string in yaml format/dict when "yaml" parameter is set, else an
+            dict is returned.
+        :rtype: string or dict
+        """
+        pod_template = {
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "leapp-{}-pod".format(self.container_name),
+                "labels": {
+                    "name": "leapp-{}-label".format(self.container_name),
+                },
+                "annotations": {
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "{}".format(self.container_name),
+                        "image": self.system.base_image(),
+                        "volumeMounts": []
+                    }
+                ],
+                "volumes": []
+            }
+        }
+
+        # Generate port list
+        if self.exposed_ports is not None and len(self.exposed_ports) > 0:
+            ports_template = {
+                "ports": []
+            }
+
+            for pmap in self.exposed_ports:
+                port = {
+                    "containerPort": pmap[1],
+                }
+                ports_template["ports"].append(port)
+
+            pod_template["spec"]["containers"][0].update(ports_template)
+
+        # Systemd additions
+        if self.system.is_systemd():
```

Comment:
We might have some problems with this approach. I'm not sure if there's something we can do about it, but I'll bring it up here for discussion.

Say there is a Centos 7 container being created here. The container will try to use the host's `/sys/fs/cgroup` directory, however, if the host (cluster node where OpenShift is running) is a Centos 6 machine, this directory won't exist there.
Another possible problem is the `hostPath`. OpenShift docs state that it won't work in a multi-node OpenShift setup (which I think is the common case). From OpenShift docs [1]:

```
This is useful for single-node development and testing only! On-host storage is not supported in any way and WILL NOT WORK in a multi-node cluster.
```
[1] https://docs.openshift.com/container-platform/3.6/rest_api/kubernetes_v1.html#v1-hostpathvolumesource